src/plugins/drawing-tool/components/drawing-layer.tsx

```diff
@@ -270,15 +286,18 @@ export class DrawingLayerView extends React.Component<DrawingLayerViewProps, Dra
     }
   }
 
-  public endSelectionBox(addToSelectedObjects: boolean) {
+  public endSelectionBox(addToSelectedObjects: boolean, e2: MouseEvent) {
     const {selectionBox} = this.state;
     if (selectionBox) {
       selectionBox.close();
-      const selectedObjects: DrawingObjectType[] = addToSelectedObjects ? this.state.selectedObjects : [];
+      let selectedObjects: DrawingObjectType[] = addToSelectedObjects ? this.state.selectedObjects : [];
       this.forEachObject((object) => {
         if (object.inSelection(selectionBox)) {
           if (selectedObjects.indexOf(object) === -1) {
             selectedObjects.push(object);
+            if (e2.metaKey || e2.shiftKey){  //when shift/meta add boxed selected objects to our state selectedObjects
+              selectedObjects = selectedObjects.concat(this.state.selectedObjects);
+            }
```

Comment:
I don't understand this If `addToSelectedObjects` is true then the selectedObjects will be the existing objects. And then the line above these changes `selectedObjects.push(object)` should be adding each of the objects in the selection box to this existing selectedObjects.
So why is it necessary to check e2.metaKey and e2.shiftKey here? Instead the code calling this `endSelectionBox` could check them and use that to decide if addToSelectedObjects is true or false.